File: Array-Hashing/longprefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = strs[0];
        int n = strs.size();
        for(int i = 1; i < n; i++){
            int j = 0;
            for(j = 0; j < min(prefix.size(), strs[i].size()); j++){
                if(prefix[j] != strs[i][j]){
                    break;
                }
            }
            prefix = prefix.substr(0, j);
        }
        return prefix;
    }
};

File: Array-Hashing/longprefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix=strs[0];
        int n=strs.length;
        int j;
        for(int i=1;i<n;i++){
            for(j = 0; j < Math.min(prefix.length(), strs[i].length()); j++){
                if(prefix.charAt(j)!=strs[i].charAt(j)){
                    break;
                }
            }
            prefix=prefix.substring(0,j);
        }
        return prefix;
    }
}